pkg/controller/controllercmd/builder.go

```diff
@@ -19,10 +20,11 @@ import (
 	leaderelectionconverter "github.com/openshift/library-go/pkg/config/leaderelection"
 	"github.com/openshift/library-go/pkg/config/serving"
 	"github.com/openshift/library-go/pkg/controller/fileobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
 )
 
 // StartFunc is the function to call on leader election start
-type StartFunc func(config *rest.Config, stop <-chan struct{}) error
+type StartFunc func(config *rest.Config, eventRecorder events.Recorder, stop <-chan struct{}) error
```

Comment:
This signature may start getting out of hand with https://github.com/openshift/library-go/pull/103 and future requirements.  Maybe refactor into a single config struct.

Reply:
/cc @deads2k 

i'm open to refactor as follow up